apps/accounts/forms.py

```diff
@@ -122,11 +122,20 @@ def grant_organisation(self, user):
 
         if Organisation.objects.filter(email__icontains=email_domain).exists():
 
-            organisation = Organisation.objects.get(email__icontains=email_domain)
-            if organisation in self.cleaned_data['chosen_organisations']:
-                org_email_domain = organisation.get_email_domain()
-                if org_email_domain == email_domain:
-                    user.approved_organisations.add(organisation)
+            # more than one org can share the same email domain
+            # find all organisations that have been selected in the form that also contain the
+            # same email domain as the user
+            organisations = Organisation.objects.filter(
+                email__icontains=email_domain,
+                pk__in=self.cleaned_data['chosen_organisations'].values_list("pk", flat=True)
+            )
+
+            # if orgs match the query, add the orgs who's email domain matches that of the user
+            if organisations.exists():
```

Comment:
I think this `.exists` adds an extra query against the db...

`if bool(organisations):` would have the same effect w/o the extra query. See the end of https://docs.djangoproject.com/en/dev/ref/models/querysets/#exists

This is really not a big deal though... I'm just in a "why do some of our views have 100+ queries on them?!" kind of mood.